contracts/DVA/DVATransferManager.sol

```diff
@@ -0,0 +1,651 @@
+
+    error SignaturesCanNotBeEmpty(bytes32 _transferID);
+
+    constructor(){
+        _txNonce = 0;
+    }
+
+    /**
+    *  @dev modify the approval criteria of a token
+     *  @param tokenAddress is the token address.
+     *  @param includeRecipientApprover determines whether the recipient is included in the approver list
+     *  @param includeAgentApprover determines whether the agent is included in the approver list
+     *  @param sequentialApproval determines whether approvals must be sequential
+     *  @param additionalApprovers are the addresses of additional approvers to be added to the approver list
+     *  Only an agent of a token can call this function
+     *  DVATransferManager must be an agent of the given token
+     *  emits an `ApprovalCriteriaModified` event
+     */
+    function modifyApprovalCriteria(
+        address tokenAddress,
+        bool includeRecipientApprover,
+        bool includeAgentApprover,
+        bool sequentialApproval,
+        address[] memory additionalApprovers
+    ) external {
+        if (!AgentRole(tokenAddress).isAgent(msg.sender)) {
+            revert OnlyTokenAgentCanCall(tokenAddress);
+        }
+
+        if (!IToken(tokenAddress).identityRegistry().isVerified(address(this))) {
+            revert DVAManagerIsNotVerifiedForTheToken(tokenAddress);
+        }
+
+        bytes32 hash = keccak256(
+            abi.encode(
+                tokenAddress,
+                includeRecipientApprover,
+                includeAgentApprover,
+                additionalApprovers
+            )
+        );
+
+        _approvalCriteria[tokenAddress] = ApprovalCriteria(
+            tokenAddress,
+            includeRecipientApprover,
+            includeAgentApprover,
+            sequentialApproval,
+            additionalApprovers,
+            hash);
+
+        emit ApprovalCriteriaModified(
+            tokenAddress,
+            includeRecipientApprover,
+            includeAgentApprover,
+            sequentialApproval,
+            additionalApprovers,
+            hash
+        );
+    }
+
+    /**
+     *  @dev initiates a new transfer
+     *  @param tokenAddress is the address of the token
+     *  @param recipient is the address of the recipient
+     *  @param amount is the transfer amount
+     *  Approval criteria must be preset for the given token address
+     *  Receiver must be verified for the given token address
+     *  emits a `TransferInitiated` event
+     */
+    function initiateTransfer(address tokenAddress, address recipient, uint256 amount) external {
+        ApprovalCriteria memory approvalCriteria = _approvalCriteria[tokenAddress];
+        if (approvalCriteria.tokenAddress == address(0)) {
+            revert TokenIsNotRegistered(tokenAddress);
+        }
+
+        IToken token = IToken(tokenAddress);
+        if (!token.identityRegistry().isVerified(recipient)) {
+            revert RecipientIsNotVerified(tokenAddress, recipient);
```

Comment:
maybe would make sense to also test `canTransfer` with address `_from` being the DvA address and `_to` the recipient for `amount`, to be sure that after approval, the transfer will be allowed to happen. Or maybe this is a bit overkill...

Reply:
We already have an e2e test for that: https://github.com/TokenySolutions/T-REX/pull/171/files#diff-5bfef1f607b3b2d5e5aa8f381f378a17e93375f7cce6bfd63dad06dc354c5b97R511